socks5-client/src/lib.rs

```diff
@@ -116,10 +116,10 @@ impl Socks5 {
                 }
                 if input[1] != 0x00 {
                     let err = ServerError::from(input[1]);
-                    *self = Socks5::Rejected(err);
-                    return Err(Error::Closed);
+                    *self = Socks5::Rejected(err, input[3], input[4]);
```

Comment:
previously if Socks5 proxy failed, we were leaving some unread data in a buffer. Not important, since the connection is useless anyway, but better to clean up!